### PR TITLE
Jwt

### PR DESCRIPTION
**Pull request by DevIgork**
i moved some dto to user and book package

### CODE REVIEW

File: src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotEmpty
+        @Size(min = 8, max = 50)
+        @Email
+        String email,
+        @NotEmpty
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java]:
```suggestion
        @NotBlank
```

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job,  but some minor change should be done!

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    null,
+                    userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request,response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("authorization");
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: Please don`t forget about constants

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            return bearerToken.substring(7);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: magic number

File: pom.xml [hunk 5]
```diff
@@ -18,6 +18,7 @@
         <java.version>17</java.version>
```
Comment (andrii-hoienko) [inline pom.xml:18]: Please resolve merge conflicts

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

**Review by DevIgork — COMMENTED** [review]
I will solve the merge config later with merges since in ide everything is ok there is just an empty line

File: src/main/resources/db/changelog/changes/07-insert-users-roles-table.yaml
Comment (d1sam) [inline src/main/resources/db/changelog/changes/07-insert-users-roles-table.yaml:1]: it's better to add `PreConditions`, example:
  ```
    preConditions:
        - sqlCheck:
            expectedResult: 1
            sql: SELECT COUNT(*) FROM users WHERE email='admin@example.com'
        - sqlCheck:
            expectedResult: 1
            sql: SELECT COUNT(*) FROM roles WHERE name='ROLE_ADMIN'
      changes:
        - insert:
            tableName: user_role
            columns:
              - column:
                  name: user_id
                  valueComputed: (SELECT id FROM users WHERE email='admin@example.com')
              - column:
                  name: role_id
                  valueComputed: (SELECT id FROM roles WHERE name='ROLE_ADMIN')
```

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, some minor change should be done, and sorry, I can't approve with conflicts :(

**Review by DevIgork — COMMENTED** [review]
sorry forgot to change the name of the commit again, these commits on top are resore conflict

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**